src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -1046,11 +1047,28 @@ public static SyntaxKind GetPreprocessorKeywordKind(string text)
             }
         }
 
+        private static SyntaxKind[] _lazyContextualKeywordKindsArray;
         public static IEnumerable<SyntaxKind> GetContextualKeywordKinds()
         {
-            for (int i = (int)SyntaxKind.YieldKeyword; i < (int)SyntaxKind.ElifKeyword ; i++)
+            if (_lazyContextualKeywordKindsArray == null)
             {
-                yield return (SyntaxKind)i;
+                ushort firstKeywordIdInclusive = (ushort)SyntaxKind.YieldKeyword;
+                ushort lastKeywordIdExclusive = (ushort)SyntaxKind.ElifKeyword;
+                var builderList = new List<SyntaxKind>(70);
```

Comment:
you could dynamically compute this with lastKeywordExclusive - firstKeywordInclusive.